packages/labs/analyzer/src/lib/javascript/modules.ts

```diff
@@ -117,7 +117,9 @@ export const getModule = (
   for (const statement of sourceFile.statements) {
     if (ts.isClassDeclaration(statement)) {
       addDeclaration(getClassDeclarationInfo(statement, analyzer));
-    } else if (ts.isFunctionDeclaration(statement)) {
+      // Ignore non-implementation signatures of overloaded functions by
+      // checking for `statement.body`.
```

Comment:
Can you maybe expand on this a bit? Like why do we want to ignore overloads? What do we do with different docs on different overloads? I don't the the CEM format explicitly handles overloads, so if we generated multiple declarations with the same name but different parameters, that would basically work by default - with the exception that tools might expect names to be unique.

I could see us documenting in the CEM that names aren't unique and duplicates indicate overloads (though we might want a way to mark and/or hide the implementation declaration because its signature might not be pretty).

Reply:
IIRC, I went down this path because there are few places in the analyzer that currently require unique names (and I just assumed this was a requirement). [Kevin's comment on my other PR](https://github.com/lit/lit/pull/3678#discussion_r1120880191) about `getField()` is related. Another example is here, where declarations in modules are collected:

https://github.com/lit/lit/blob/be18292938062a3b5233016fdac1a72ba6f1eacf/packages/labs/analyzer/src/lib/javascript/modules.ts#L105-L107

I think your suggestion of explicitly documenting that names of potentially overloadable items in collections are not necessarily unique sounds like the best way to go. That way I could remove the weird documentation merging stuff here and we would avoid losing information in the output. Conveniently, the CEM schema already seems prepared for this since most (all?) collections of declarations are arrays of items with `name` properties and not an object / map that requires a unique key per item. This would also resolve the issue you pointed out in your other comment about this PR not matching TS's behavior around docs for overloads.

Reply:
which would move the responsibility for matching TS's behavior or not to consumers of the manifest. That sounds good to me.